test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -10151,6 +10151,47 @@ BOOST_AUTO_TEST_CASE(constant_string)
 	ABI_CHECK(callContractFunction("h()"), encodeDyn(string("hello")));
 }
 
+BOOST_AUTO_TEST_CASE(abi_encode)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f() returns (bytes) {
+				return abi.encode(1, 2);
+			}
+			function g() returns (bytes) {
+				return abi.encodePacked(uint256(1), uint256(2));
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(u256(32), u256(64), u256(1), u256(2)));
+	ABI_CHECK(callContractFunction("g()"), encodeArgs(u256(32), u256(64), u256(1), u256(2)));
+}
+
+BOOST_AUTO_TEST_CASE(abi_encode_call)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(uint8 a, string b) {
+				require(a == 42);
+				require(b.length == 2);
+				require(bytes(b)[0] == 72); // 'H'
+				require(bytes(b)[1] == 101); // 'e'
+			}
+			function g(uint8 a) {
+				bytes request = abi.encode(
+					bytes4(keccak256(abi.encodePacked("f(uint8,string)"))),
+					a,
+					"He"
+				);
+				require(this.call(request) == true);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	ABI_CHECK(callContractFunction("g(uin8)"), encodeArgs(u256(42)));
```

Comment:
Missing a t in uint8

Reply:
Also the `encodeArgs` call has to be moved to an argument to callContractFunction, or, even shorter:
`ABI_CHECK(callContractFunction("g(uin8)", 42), bytes());`